src/processor.cpp

```diff
@@ -0,0 +1,416 @@
+#include "game.h"
+#include "map.h"
+#include "mapdata.h"
+#include "generic_factory.h"
+#include "messages.h"
+#include "json.h"
+#include "vehicle.h"
+#include "processor.h"
+
+#include <string>
+
+// Processor namespace
+
+namespace
+{
+
+struct processor_data;
+using processor_id = string_id<processor_data>;
+
+struct processor_data {
+
+    processor_data() :
+        was_loaded( false ) {};
+
+    processor_id id; //furniture
+    furn_id next_type;
```

Comment:
`furn_id` is an `int_id` - this one can't be loaded from a string directly. You'll need to load it as `furn_str_id`.